### PR TITLE
Issue #6121: Update FullIdent to store DetailAST instead of line/column

### PR DESCRIPTION
**Pull request by pbludov**
Issue #6121 

Note that the whole FullIdent::append was changed to simple
```
        if (detailAST == null) {
            detailAST = ast;
        }
```

this is because the deepest `IDENT` in the chain as always the top- and leftmost one.

### CODE REVIEW

File: src/test/java/com/puppycrawl/tools/checkstyle/api/FullIdentTest.java [hunk 1]
```diff
@@ -47,6 +47,13 @@ public void testToString() {
         Assert.assertEquals("Invalid full indent", "MyTest[15x14]", indent.toString());
     }
 
+    @Test
+    public void testGetDetailAst() {
+        final DetailAST ast = new DetailAST();
+        final FullIdent indent = FullIdent.createFullIdent(ast);
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/api/FullIdentTest.java]: This doesn't seem like a very detailed test unless we have one that is more complex than this. Please verify.
Can we have a test with more then 1 AST to show we stick with only the first one even if there is more.

File: src/main/java/com/puppycrawl/tools/checkstyle/api/FullIdent.java [hunk 2]
```diff
@@ -77,25 +75,34 @@ public String getText() {
         return String.join("", elements);
     }
 
+    /**
+     * Gets the topmost leftmost DetailAST for this FullIdent.
+     * @return the topmost leftmost ast
+     */
+    public DetailAST getDetailAST() {
```
[the remark this answers is not in the document]
Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/FullIdent.java]: As another note, I am not sure this method name is detailed enough to know what it is for without looking at javadoc.
@romani Do you have an opinion?

File: src/main/java/com/puppycrawl/tools/checkstyle/api/FullIdent.java [hunk 3]
```diff
@@ -40,10 +40,8 @@
 
     /** The list holding subsequent elements of identifier. **/
     private final List<String> elements = new ArrayList<>();
-    /** The line number. **/
-    private int lineNo;
-    /** The column number. **/
-    private int columnNo;
+    /** The topmost and leftmost AST of the fullIdent. */
+    private DetailAST detailAST;
```
[the remark this answers is not in the document]
Comment (pbludov) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/FullIdent.java]: The name should match [SimpleAccessorNameNotationCheck](http://sevntu-checkstyle.github.io/sevntu.checkstyle/apidocs/com/github/sevntu/checkstyle/checks/coding/SimpleAccessorNameNotationCheck.html) policy.

File: src/test/java/com/puppycrawl/tools/checkstyle/api/FullIdentTest.java (hunk 1 above)
Comment (pbludov) [reply src/test/java/com/puppycrawl/tools/checkstyle/api/FullIdentTest.java]: done

File: src/main/java/com/puppycrawl/tools/checkstyle/api/FullIdent.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (pbludov) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/FullIdent.java]: Changed to detailAst. If the CI will be red, i'll change it back.

[the remark this answers is not in the document]
Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/FullIdent.java]: @pbludov That check is just enforcing the name of the getter/setter match the field name. If you change the field name, you have to rename the method.

**Comment by pbludov** [comment]
Regression report:
https://pbludov.github.io/issue-6121/

**Comment by rnveach** [comment]
@pbludov 
The report shows differences when none are expected.
https://pbludov.github.io/issue-6121/checkstyle/index.html#A1
It looks like old violation is based on line 7 instead of line where violation is reported, line 6.

The change in violation looks correct to me. We will need to add it to our tests.

**Comment by pbludov** [comment]
> Please look over the report before and point out anything that isn't expected.
> https://pbludov.github.io/issue-6121/checkstyle/index.html#A1

I do not see anything wrong with this diff. The [old code](https://github.com/checkstyle/checkstyle/pull/6122/files#diff-e379f4a4f12f2d9d65aefd8c9ef0bff7L138) calculates the line and the column in a crazy way:
```
import java .       << min line
net.Socket    ;
    ^ min nonzero col
```
Which gives
```
      Ident start here
     V
import java .
net.Socket    ;
```
Thats it.

**Comment by rnveach** [comment]
> nonzero col

That explains it. I didn't realize there was a check for non-zero.

Please add these to the tests for UnusedImports.

**Comment by pbludov** [comment]
> Please add these to the tests for UnusedImports.

done.

**Comment by rnveach** [comment]
@romani I am fine with PR, please review. Issue was never approved but it was broken off from an approved issue so I don't see a problem there.

**Comment by romani** [comment]
Issue is approved, Travis vis restarted, there was not only PR verification failure, so we need to make Travis green

**Pull request merged**